src/shell.cpp

```diff
@@ -0,0 +1,58 @@
+#include "shell.h"
+
+void Shell::runCommand(const std::string& command, bool inBackground) {
+    std::vector<std::string> args = parser.split(command, ' ');
+    std::vector<char*> cargs;
+    for (const auto& arg : args) {
+        cargs.push_back(const_cast<char*>(arg.c_str()));
+    }
+    cargs.push_back(NULL);
+
+    pid_t pid = fork();
+
+    if (pid == -1) {
+        std::cerr << "Failed to fork.\n";
+        return;
+    }
+    else if (pid > 0) {
+        // In parent process
+        if (inBackground) {
+            manager.addProcess(pid, args[0]);
+        }
+        else {
+            waitpid(pid, NULL, 0);
+        }
+    }
+    else {
+        // In child process
+        if (execvp(cargs[0], &cargs[0]) == -1) {
```

Comment:
`execvp` finds the executable file for you, but you were asked to implement that by yourself